### PR TITLE
[#7] Fix the origin on TimeoutExceptions

### PR DESCRIPTION
**Pull request by jkenny**

### CODE REVIEW

File: src/main/scala/com/gilt/gfc/concurrent/FutureBuilder.scala [hunk 1]
```diff
@@ -78,14 +79,14 @@ case class FutureBuilder[A,R] private (
     *         Future[Try[SomeServiceCallResult]] in case there's no default,
     *         a 'checked exception' of sorts.
     */
-  def runWithTimeout( after: FiniteDuration
+  def runWithTimeout( after: FiniteDuration, errorMessage: Option[String] = None
```
Comment (gheine) [inline src/main/scala/com/gilt/gfc/concurrent/FutureBuilder.scala]: In order to preserve backward compatibility, please revert this and add a new/second function that has a required error message parameter of type String

File: src/main/scala/com/gilt/gfc/concurrent/ScalaFutures.scala [hunk 2]
```diff
@@ -20,8 +20,8 @@ object ScalaFutures {
     /**
      * Create a new Future that times out with a [[java.util.concurrent.TimeoutException]] after the given FiniteDuration
      */
-    def withTimeout(after: FiniteDuration)(implicit ec: ExecutionContext): Future[A] =
-      Future.firstCompletedOf(Seq(f, Timeouts.timeout(after)))
+    def withTimeout(after: FiniteDuration, errorMessage: Option[String] = None)(implicit ec: ExecutionContext): Future[A] =
```
Comment (gheine) [inline src/main/scala/com/gilt/gfc/concurrent/ScalaFutures.scala]: In order to preserve backward compatibility, please add a new/second function that has a required error message parameter of type String

**Comment by gheine** [comment]
Looks good, some comments inline.

**Comment by jkenny** [comment]
Do we need to worry about backwards compatibility? The library is only at version 0.x

**Comment by gheine** [comment]
:+1:

**Pull request merged**